griptape/drivers/embedding/base_embedding_driver.py

```diff
@@ -1,21 +1,62 @@
+from __future__ import annotations
+import numpy as np
 from abc import ABC, abstractmethod
 from attr import define, field
 from griptape.artifacts import TextArtifact
 from griptape.mixins import ExponentialBackoffMixin
+from griptape.tokenizers import BaseTokenizer
+from griptape.chunkers import BaseChunker, TextChunker
 
 
 @define
 class BaseEmbeddingDriver(ExponentialBackoffMixin, ABC):
     dimensions: int = field(kw_only=True)
+    tokenizer: BaseTokenizer = field(kw_only=True)
+    chunker: BaseChunker = field(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        self.chunker = TextChunker(tokenizer=self.tokenizer)
```

Comment:
Let's move this to a `Factory`.

Reply:
Had it as a `Factory` originally but was running into errors about `tokenizer` not being initialized. Let me try again.

Reply:
I think it might have to be `kw_only=True`.

Reply:
Removed default, child classes must initialize the Chunker.

Reply:
Kept in `__attrs_post_init` based on Slack conversation.